public/src/components/channelManagement/variantCloneButton.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import { useForm } from 'react-hook-form';
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  IconButton,
+  makeStyles,
+  TextField,
+  Theme,
+} from '@material-ui/core';
+
+import CloseIcon from '@material-ui/icons/Close';
+
+import { Variant } from './helpers/shared';
+
+import useOpenable from '../../hooks/useOpenable';
+
+import {
+  EMPTY_ERROR_HELPER_TEXT,
+  INVALID_CHARACTERS_ERROR_HELPER_TEXT,
+  VALID_CHARACTERS_REGEX,
+  createDuplicateValidator,
+} from './helpers/validation';
+
+const useStyles = makeStyles(({ spacing }: Theme) => ({
```

Comment:
this file copies a lot of css and logic from `testNewVariantButton.tsx`. We can refactor slightly to share a `createVariantDialog.tsx`, like we do for tests [here](https://github.com/guardian/support-admin-console/blob/main/public/src/components/channelManagement/createTestDialog.tsx#L69).
I've pushed a branch to show how this could be done: https://github.com/guardian/support-admin-console/pull/301

Reply:
Done